src/main/java/seedu/address/storage/JsonSerializableTutorsPet.java

```diff
@@ -54,6 +61,13 @@ public TutorsPet toModelType() throws IllegalValueException {
             }
             tutorsPet.addStudent(student);
         }
+        for (JsonAdaptedModuleClass jsonAdaptedModuleClass : classes) {
+            ModuleClass moduleClass = jsonAdaptedModuleClass.toModelType();
+            if (tutorsPet.hasModuleClass(moduleClass)) {
+                throw new IllegalValueException(MESSAGE_DUPLICATE_MODULE_CLASS);
+            }
+            tutorsPet.addModuleClass(moduleClass);
+        }
```

Comment:
Should this for loop be placed in another method for proper SRP? The for loop above that iterates through `jsonAdaptedStudent`, can be similarly placed in another method. Then, this `toModelType()` method will call these two for loop methods.

Reply:
Resolved by abstracting code inside toModelType() into two smaller methods.